### PR TITLE
[Arm64]:Fixing the Logic of ResolveWorkerChainLookupAsmStub for VSD

### PR DESCRIPTION
**Pull request by ramarag**
@rahku @gkhanna79 PTAL

@dotnet/arm64-contrib : FYI

### CODE REVIEW

**Comment by ramarag** [comment]
@dotnet-bot test  Windows_NT arm64

**Comment by ramarag** [comment]
@dotnet-bot test Windows_NT arm64 chk

**Comment by kyulee1** [comment]
@dotnet-bot test Windows_NT arm64 Release

**Comment by kyulee1** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by kyulee1** [comment]
@dotnet-bot test Windows_NT arm64 Debug

File: src/vm/arm64/asmhelpers.asm [hunk 1]
```diff
@@ -1121,12 +1121,12 @@ MainLoop
         cmp     x9, #0
         beq     Fail
 
-        ldr     x9,  [x9, #0]
-        cmp     x9, x13           ; compare our MT with the one in the ResolveCacheElem
+        ldr     x16, [x9, #0]
```
Comment (briansull) [inline src/vm/arm64/asmhelpers.asm]: If you have a second register available you can use a ldp instruction:

ldp   x16,x17, [x9]

you don't need the ", #0" as that is implied by default.

Comment (briansull) [inline src/vm/arm64/asmhelpers.asm]: Also update the comment in the header to mention that x16 and/or x17 are modified by this method.

The registers x16 and x17 are also known as ip0 and ip1

Comment (briansull) [inline src/vm/arm64/asmhelpers.asm]: LGTM with comments

**Comment by rahku** [comment]
Looks good. It might be beneficial to do some unit testing of VSD now that jit is up.

**Comment by ramarag** [comment]
I have tried a few scenarios but the failure i am addressing does not come up easily.
There are a few unit tests for VSD already checked in https://github.com/dotnet/coreclr/tree/master/tests/src/JIT/opt/virtualstubdispatch

**Comment by ramarag** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Pull request merged**